### PR TITLE
Fix name badge build, remove cycle dependencies.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | https://github.com/yiisoft/db/issues/105

- [x] Remove dependency cycle composer.
- [x] Fix badged name build.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,20 +20,23 @@
...
         "ext-json": "*",
-        "yiisoft/db": "^3.0@dev",
         "yiisoft/arrays": "^3.0@dev",
         "yiisoft/strings": "^3.0@dev"
     },
     "require-dev": {
         "hiqdev/composer-config-plugin": "^1.0@dev",
         "phpunit/phpunit": "^9.0",
         "yiisoft/aliases": "^3.0@dev",
+        "yiisoft/db": "^3.0@dev",
```
Comment (schmunk42) [inline composer.json]: Why is this moved to `require-dev` isn't it mandatory?

Comment (terabytesoftw) [reply composer.json]: If you are right and for that reason it was placed required, but when the package is initially separated it has a cyclic reference, which does not allow running the tests in db, we need to develop the packages, change many public properties to private ones, and change the structure of the same, so that they can be independent, then the idea is to make the changes, in this first stage it was possible to pass all the tests.

Comment (schmunk42) [reply composer.json]: Are these the cyclic references? https://github.com/yiisoft/db/blob/aad933b2b33f7ae210e10c4c4fc7dea03a542a2e/composer.json#L35-L37

Comment (terabytesoftw) [reply composer.json]: If these are correct, they are necessary to verify the tests, since any change in db or in each package affects it as a whole, in yii2 there is no problem because the package is together.

Comment (schmunk42) [reply composer.json]: But tests for `db-pgsql` running in `yii-db` are merely integration tests, they should not affect development of `yii-db` otherwise it would make no sense to split the packages.

Why don't we install `db-mysql`, `db-pgsql` manually when required for integration testing? Instead of *suggesting* a *mandatory* package in those repos.

Comment (terabytesoftw) [reply composer.json]: done.
